java/arcs/core/data/SchemaFields.kt

```diff
@@ -56,18 +56,27 @@ sealed class FieldType(
     }
 }
 
-/** Arcs primitive types. */
-enum class PrimitiveType {
-    Boolean,
-    Number,
-    Text,
-    Byte,
-    Short,
-    Int,
-    Long,
-    Char,
-    Float,
-    Double
+/**
+ * Arcs primitive types.
+ *
+ * Reordering the ids in this enum will require a DB migration. It is safe to append
+ * additional entries, up to the limit of [DatabaseImpl.REFERENCE_TYPE_SENTINEL].
+ * Note that the order of appearance in this list doesn't matter, just the values
+ * associated with each PrimitiveType.
+ *
+ * When adding new entries, ensure each PrimitiveType gets assigned a unique id!
```

Comment:
it would be good to add a test that verifies all ids are different

Reply:
A note about adding a new primitive type requiring a migration would also be good.  (since the primitive types are only added in `onCreate`)

Reply:
Done and done!